src/compiler/checker.ts

```diff
@@ -31208,8 +31208,10 @@ namespace ts {
         }
 
         function isParameterDeclarationWithIdentifierName(symbol: Symbol) {
-            return symbol.valueDeclaration && isParameter(symbol.valueDeclaration) && isIdentifier(symbol.valueDeclaration.name);
+            const declaration = symbol.valueDeclaration;
+            return declaration && isParameter(declaration) && declaration.name && isIdentifier(declaration.name);

```

Comment:
Isn't this just monkey-patching around the actual issue? According to the type definition this property should not be falsy, and `strictNullChecks` are enabled. That means something else is going wrong before already, and this check is just hiding the actual bug.

Reply:
Why did you mark this conversation as resolved? I'm confused.

Reply:
@MartinJohns I'm ok to see your PR, that resolves the "actual" issue. I'll close this PR.